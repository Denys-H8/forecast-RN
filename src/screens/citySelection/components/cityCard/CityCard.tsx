import React from 'react';
import { View, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Fontisto';
import CustomText from 'src/components/text';
import constants from 'src/constants';
import { getWeatherIconName } from 'src/helpers';
import { TCity, TGroupWeatherItem } from 'src/store/citySelection/reducer';
import { styles } from './cityCard.styles';

type Props = {
  item: TGroupWeatherItem;
  onPress: (city: TCity) => void;
};

const CityCard: React.FC<Props> = ({ item, onPress }) => {
  const city = { id: item.id, name: item.name, lat: item.coord.lat, lon: item.coord.lon };

  return (
    <TouchableOpacity style={styles.itemContainer} testID="CityCard.TouchableOpacity" onPress={() => onPress(city)}>
      <View style={styles.degreesContainer}>
        <CustomText style={styles.number}>{item.main.temp.toFixed(1)}</CustomText>
        <CustomText style={styles.celsius}>0</CustomText>
      </View>
      <View style={styles.cityWeatherContainer}>
        <View style={styles.iconContainer}>
          <Icon
            name={getWeatherIconName(item.weather[0].description)}
            size={35}
            color={constants.SUB_COLOR}
            style={styles.icon}
          />
        </View>
        <View style={styles.cityContainer}>
          <CustomText style={styles.cityName}>{item.name}</CustomText>
          <CustomText style={styles.weatherName}>{item.weather[0].description.toUpperCase()}</CustomText>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default CityCard;
