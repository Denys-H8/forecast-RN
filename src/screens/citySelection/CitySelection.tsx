import React, { useCallback, useEffect } from 'react';
import { FlatList, View, RefreshControl } from 'react-native';
import CityCard from './components/cityCard';
import SearchField from './components/searchField';
import { cities } from './cityData';
import { styles } from './citySelection.styles';
import { useDispatch, useSelector } from 'react-redux';
import { cityActions } from 'src/store/rootActions';
import { debounce } from 'lodash';
import * as CITY_SELECTORS from '../../store/citySelection/selectors';
import { TData, TGroupWeatherItem, TGroupWeather, TCity } from 'src/store/citySelection/reducer';

const CitySelection: React.FC = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(cityActions.resetQuery());
    dispatch(cityActions.getWeatherForCities());
  }, [dispatch]);

  const query = useSelector(CITY_SELECTORS.query);
  const isLoading = useSelector(CITY_SELECTORS.isLoadingSelector);

  const handleSearch = (query: string) => {
    const data: TData[] = cities.filter((item: TCityBig) => {
      return item.name.toLowerCase().includes(query.toLowerCase());
    });
    dispatch(cityActions.queryChanged(query, data));
    handleSearchSaga();
  };

  const handleSearchSaga = useCallback(
    debounce(() => {
      dispatch(cityActions.setLoading(true));
      dispatch(cityActions.getWeatherForCities());
    }, 700),
    [],
  );

  const handleCityPress = (city: TCity) => {
    dispatch(cityActions.pressedCity(city));
  };

  const keyExtractor = (item: TGroupWeatherItem) => item.id.toString();
  const listHeaderComponent = <SearchField onChangeText={handleSearch} value={query} />;
  const itemSeparatorComponent = () => <View style={styles.separator} />;
  const renderItem = ({ item }: any) => <CityCard item={item} onPress={handleCityPress} />;
  const weatherData: TGroupWeather = useSelector(CITY_SELECTORS.weatherData);

  return (
    <FlatList
      style={styles.container}
      keyExtractor={keyExtractor}
      data={weatherData.list}
      ListHeaderComponent={listHeaderComponent}
      stickyHeaderIndices={[0]}
      ItemSeparatorComponent={itemSeparatorComponent}
      renderItem={renderItem}
      refreshControl={<RefreshControl refreshing={isLoading} />}
    />
  );
};

type TCityBig = {
  id: number;
  name: string;
  state: string;
  country: string;
  coord: {
    lon: number;
    lat: number;
  };
};

export default CitySelection;
