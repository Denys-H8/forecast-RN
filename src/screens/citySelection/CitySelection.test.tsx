import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import CitySelection from './CitySelection';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { itemMock } from 'src/screens/citySelection/components/cityCard/CityCard.test';
import {
  queryChanged,
  resetQuery,
  getWeatherForCities,
  setLoading,
  pressedCity,
} from 'src/store/citySelection/actions';

const mockStore = configureStore([]);
const city = {
  id: 1024546,
  lat: 1024546,
  lon: 1525545,
  name: 'City',
};
const store = mockStore({
  cityReducer: {
    query: '',
    isLoading: false,
    weatherData: {
      list: [itemMock],
      city: {
        id: 1341,
        name: 'name',
        coord: {
          lat: '445154',
          lon: '414154',
        },
        sunrise: 551454,
        sunset: 411411,
      },
    },
  },
});

describe('CitySelection ', () => {
  test('should render', () => {
    const { toJSON } = render(
      <Provider store={store}>
        <CitySelection></CitySelection>
      </Provider>,
    );
    expect(toJSON()).toMatchSnapshot();
  });

  test('should handle search', async () => {
    const newTextValue = 'AVACD';
    store.dispatch = jest.fn();

    const { toJSON, getByTestId } = render(
      <Provider store={store}>
        <CitySelection></CitySelection>
      </Provider>,
    );

    fireEvent(getByTestId('SearchField.TextInput'), 'onChangeText', newTextValue);

    await new Promise((r) => setTimeout(r, 2000));

    expect(toJSON()).toMatchSnapshot();
    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledTimes(5);
    expect(store.dispatch).toHaveBeenCalledWith(resetQuery());
    expect(store.dispatch).toHaveBeenCalledWith(getWeatherForCities());
    expect(store.dispatch).toHaveBeenCalledWith(queryChanged(newTextValue, []));
    expect(store.dispatch).toHaveBeenCalledWith(setLoading(true));
    expect(store.dispatch).toHaveBeenCalledWith(getWeatherForCities());
  });

  test('should handle city press', async () => {
    store.dispatch = jest.fn();

    const { toJSON, getByTestId } = render(
      <Provider store={store}>
        <CitySelection></CitySelection>
      </Provider>,
    );

    fireEvent.press(getByTestId('CityCard.TouchableOpacity'));

    expect(toJSON()).toMatchSnapshot();
    expect(store.dispatch).toHaveBeenCalled();
    expect(store.dispatch).toHaveBeenCalledTimes(3);
    expect(store.dispatch).toHaveBeenCalledWith(resetQuery());
    expect(store.dispatch).toHaveBeenCalledWith(getWeatherForCities());
    expect(store.dispatch).toHaveBeenCalledWith(pressedCity(city));
  });
});
