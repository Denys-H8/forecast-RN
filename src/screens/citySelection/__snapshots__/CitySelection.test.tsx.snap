// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityCard 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "justifyContent": "space-between",
      "opacity": 1,
      "paddingLeft": 33,
      "paddingVertical": 20,
    }
  }
  testID="CityCard.TouchableOpacity"
>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flexDirection": "row",
        "width": 80,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 36,
            "lineHeight": 42,
          },
        ]
      }
    >
      20.7
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "WorkSans-Regular",
          },
          Object {
            "color": "#fff",
            "fontSize": 18,
            "lineHeight": 18,
          },
        ]
      }
    >
      0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "width": 80,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#8B8EA5",
              "fontSize": 35,
            },
            Object {
              "marginRight": 28,
            },
            Object {
              "fontFamily": "Fontisto",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "WorkSans-Regular",
            },
            Object {
              "color": "#fff",
              "fontFamily": "WorkSans-SemiBold",
              "fontSize": 24,
              "lineHeight": 28,
            },
          ]
        }
      >
        City
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "WorkSans-Regular",
            },
            Object {
              "color": "#8B8EA5",
              "fontSize": 12,
              "lineHeight": 14,
            },
          ]
        }
      >
        SUNNY
      </Text>
    </View>
  </View>
</View>
`;

exports[`CitySelection  should handle city press 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <SearchField
      onChangeText={[Function]}
      value=""
    />
  }
  data={
    Array [
      Object {
        "clouds": Object {
          "all": 20,
        },
        "coord": Object {
          "lat": 1024546,
          "lon": 1525545,
        },
        "dt": 1024546,
        "id": 1024546,
        "main": Object {
          "feels_like": 23,
          "temp": 20.7,
          "temp_max": 21,
          "temp_min": 15,
        },
        "name": "City",
        "sys": Object {
          "sunrise": 1525545,
          "sunset": 1024546,
        },
        "visibility": 95,
        "weather": Array [
          Object {
            "description": "Sunny",
          },
        ],
        "wind": Object {
          "deg": 10,
          "speed": 10,
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  style={
    Object {
      "backgroundColor": "#191C39",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#191C39",
            "paddingBottom": 30,
            "paddingHorizontal": 32,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#fff",
              "borderBottomWidth": 2,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Enter the name of city"
            placeholderTextColor="#8B8EA5"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#fff",
                "flex": 1,
                "fontFamily": "WorkSans-Regular.ttf",
                "fontSize": 16,
                "marginRight": 10,
              }
            }
            testID="SearchField.TextInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 40,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingLeft": 33,
            "paddingVertical": 20,
          }
        }
        testID="CityCard.TouchableOpacity"
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "width": 80,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 36,
                  "lineHeight": 42,
                },
              ]
            }
          >
            20.7
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "lineHeight": 18,
                },
              ]
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "width": 80,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 35,
                  },
                  Object {
                    "marginRight": 28,
                  },
                  Object {
                    "fontFamily": "Fontisto",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": "WorkSans-SemiBold",
                    "fontSize": 24,
                    "lineHeight": 28,
                  },
                ]
              }
            >
              City
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 12,
                    "lineHeight": 14,
                  },
                ]
              }
            >
              SUNNY
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`CitySelection  should handle search 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <SearchField
      onChangeText={[Function]}
      value=""
    />
  }
  data={
    Array [
      Object {
        "clouds": Object {
          "all": 20,
        },
        "coord": Object {
          "lat": 1024546,
          "lon": 1525545,
        },
        "dt": 1024546,
        "id": 1024546,
        "main": Object {
          "feels_like": 23,
          "temp": 20.7,
          "temp_max": 21,
          "temp_min": 15,
        },
        "name": "City",
        "sys": Object {
          "sunrise": 1525545,
          "sunset": 1024546,
        },
        "visibility": 95,
        "weather": Array [
          Object {
            "description": "Sunny",
          },
        ],
        "wind": Object {
          "deg": 10,
          "speed": 10,
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  style={
    Object {
      "backgroundColor": "#191C39",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#191C39",
            "paddingBottom": 30,
            "paddingHorizontal": 32,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#fff",
              "borderBottomWidth": 2,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Enter the name of city"
            placeholderTextColor="#8B8EA5"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#fff",
                "flex": 1,
                "fontFamily": "WorkSans-Regular.ttf",
                "fontSize": 16,
                "marginRight": 10,
              }
            }
            testID="SearchField.TextInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 40,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingLeft": 33,
            "paddingVertical": 20,
          }
        }
        testID="CityCard.TouchableOpacity"
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "width": 80,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 36,
                  "lineHeight": 42,
                },
              ]
            }
          >
            20.7
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "lineHeight": 18,
                },
              ]
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "width": 80,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 35,
                  },
                  Object {
                    "marginRight": 28,
                  },
                  Object {
                    "fontFamily": "Fontisto",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": "WorkSans-SemiBold",
                    "fontSize": 24,
                    "lineHeight": 28,
                  },
                ]
              }
            >
              City
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 12,
                    "lineHeight": 14,
                  },
                ]
              }
            >
              SUNNY
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`CitySelection  should render 1`] = `
<RCTScrollView
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <SearchField
      onChangeText={[Function]}
      value=""
    />
  }
  data={
    Array [
      Object {
        "clouds": Object {
          "all": 20,
        },
        "coord": Object {
          "lat": 1024546,
          "lon": 1525545,
        },
        "dt": 1024546,
        "id": 1024546,
        "main": Object {
          "feels_like": 23,
          "temp": 20.7,
          "temp_max": 21,
          "temp_min": 15,
        },
        "name": "City",
        "sys": Object {
          "sunrise": 1525545,
          "sunset": 1024546,
        },
        "visibility": 95,
        "weather": Array [
          Object {
            "description": "Sunny",
          },
        ],
        "wind": Object {
          "deg": 10,
          "speed": 10,
        },
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  style={
    Object {
      "backgroundColor": "#191C39",
      "flex": 1,
    }
  }
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Object {
            "backgroundColor": "#191C39",
            "paddingBottom": 30,
            "paddingHorizontal": 32,
            "paddingTop": 15,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderBottomColor": "#fff",
              "borderBottomWidth": 2,
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            onChangeText={[Function]}
            placeholder="Enter the name of city"
            placeholderTextColor="#8B8EA5"
            rejectResponderTermination={true}
            style={
              Object {
                "color": "#fff",
                "flex": 1,
                "fontFamily": "WorkSans-Regular.ttf",
                "fontSize": 16,
                "marginRight": 10,
              }
            }
            testID="SearchField.TextInput"
            underlineColorAndroid="transparent"
            value=""
          />
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontSize": 40,
                },
                undefined,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingLeft": 33,
            "paddingVertical": 20,
          }
        }
        testID="CityCard.TouchableOpacity"
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "width": 80,
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 36,
                  "lineHeight": 42,
                },
              ]
            }
          >
            20.7
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontFamily": "WorkSans-Regular",
                },
                Object {
                  "color": "#fff",
                  "fontSize": 18,
                  "lineHeight": 18,
                },
              ]
            }
          >
            0
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "justifyContent": "flex-start",
            }
          }
        >
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "center",
                "width": 80,
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 35,
                  },
                  Object {
                    "marginRight": 28,
                  },
                  Object {
                    "fontFamily": "Fontisto",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "flex-start",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#fff",
                    "fontFamily": "WorkSans-SemiBold",
                    "fontSize": 24,
                    "lineHeight": 28,
                  },
                ]
              }
            >
              City
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "fontFamily": "WorkSans-Regular",
                  },
                  Object {
                    "color": "#8B8EA5",
                    "fontSize": 12,
                    "lineHeight": 14,
                  },
                ]
              }
            >
              SUNNY
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
